### PR TITLE
Fix some bad code by making it worse

### PR DESCRIPTION
**Pull request by tavplubix**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

**Comment by robot-ch-test-poll4** [comment]
*This is an automated comment for commit 0f4260bf2694125b64e89ffbd45c1a9143533bb3 with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/56026/0f4260bf2694125b64e89ffbd45c1a9143533bb3/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td><a href="https://github.com/ClickHouse/ClickHouse/actions/runs/6644226193">✅ success</a></td></tr>
<tbody>
</table>
</details>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/56026/0f4260bf2694125b64e89ffbd45c1a9143533bb3/ci_running.html">⏳ pending</a></td></tr>
<tbody>
</table>

**Review by alesapin — APPROVED** [review]

File: src/Databases/DatabaseReplicated.cpp [hunk 1]
```diff
@@ -585,7 +585,8 @@ void DatabaseReplicated::checkQueryValid(const ASTPtr & query, ContextPtr query_
 
         if (auto * create = query->as<ASTCreateQuery>())
         {
-            bool replicated_table = create->storage && create->storage->engine && startsWith(create->storage->engine->name, "Replicated");
+            bool replicated_table = create->storage && create->storage->engine &&
```
Comment (alexey-milovidov) [inline src/Databases/DatabaseReplicated.cpp:588]: Why don't add a method `isReplicated` to `IStorage`?

Comment (tavplubix) [reply src/Databases/DatabaseReplicated.cpp:588]: There's no instance of `IStorage` here. But maybe we can add it to `StorageFactory`

**Comment by tavplubix** [comment]
Integration tests (asan) [6/6]  - it failed to fetch docker images
Integration tests (tsan) [4/6]  - same

**Pull request merged**